contrib/upgradability-patterns.asciidoc

```diff
@@ -0,0 +1,353 @@
+  }
+
+  /**
+   * @dev delete the value inside StringValues for the given key 
+   * @param record sha3 key
+   */
+  function deleteStringValue(bytes32 record) public onlyOwner {
+    delete StringValues[record];
+  }
+
+  /**
+   * @dev change the value inside AddressValues for the given key 
+   * @param record sha3 key
+   * @param value new value is address
+   */
+  function setAddressValue(bytes32 record, address value) public onlyOwner {
+    AddressValues[record] = value;
+  }
+
+  /**
+   * @dev delete the value inside AddressValues for the given key 
+   * @param record sha3 key
+   */
+  function deleteAddressValue(bytes32 record) public onlyOwner {
+    delete AddressValues[record];
+  }
+
+  /**
+   * @dev change the value inside BytesValues for the given key 
+   * @param record sha3 key
+   * @param value new value is bytes
+   */
+  function setBytesValue(bytes32 record, bytes value) public onlyOwner {
+    BytesValues[record] = value;
+  }
+
+  /**
+   * @dev delete the value inside BytesValues for the given key 
+   * @param record sha3 key
+   */
+  function deleteBytesValue(bytes32 record) public onlyOwner {
+    delete BytesValues[record];
+  }
+
+  /**
+   * @dev change the value inside Bytes32Values for the given key 
+   * @param record sha3 key
+   * @param value new value is bytes32
+   */
+  function setBytes32Value(bytes32 record, bytes32 value) public onlyOwner {
+    Bytes32Values[record] = value;
+  }
+
+  /**
+   * @dev delete the value inside Bytes32Values for the given key 
+   * @param record sha3 key
+   */
+  function deleteBytes32Value(bytes32 record) public onlyOwner {
+    delete Bytes32Values[record];
+  }
+
+  /**
+   * @dev change the value inside BooleanValues for the given key 
+   * @param record sha3 key
+   * @param value new value is bool
+   */
+  function setBooleanValue(bytes32 record, bool value) public onlyOwner {
+    BooleanValues[record] = value;
+  }
+
+  /**
+   * @dev delete the value inside BooleanValues for the given key 
+   * @param record sha3 key 
+   */
+  function deleteBooleanValue(bytes32 record) public onlyOwner {
+    delete BooleanValues[record];
+  }
+}
+----
+
+*UpgradableContract*
+----
+pragma solidity ^0.4.13;
+
+import '../math/SafeMath.sol';
+import "../data/EternalStorage.sol";
+
+library CounterLibrary {
+  using SafeMath for uint256;
+
+  function getCount(address _storageContract) public constant returns(uint256) {
+    return EternalStorage(_storageContract).UIntValues(sha3("counter"));
+  } 
+    
+  function increment(address _storageContract) public returns (uint) {
+    uint count = getCount(_storageContract);
+    uint value = count.add(1);
+    EternalStorage(_storageContract).setUIntValue(sha3("counter"), value);
+    return value;
+  }
+} 
+----
+
+----
+pragma solidity ^0.4.13;
+
+import "../lifecycle/Pausable.sol";
+import "../examples/SampleLibraryEternalStorage.sol";
+
+contract SampleContractWithEternalStorage is Pausable {
```

Comment:
Also, an example of a newer version of the contract using the storage contract would make it clear to the reader.